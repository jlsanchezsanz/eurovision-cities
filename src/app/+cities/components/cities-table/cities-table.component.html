<div class="mat-elevation-z8">
  <!-- Filter -->
  <section>
    <mat-form-field>
      <input
        matInput
        (keyup)="applyFilter($event.target.value)"
        placeholder="Filter"
      />
    </mat-form-field>
  </section>

  <!-- Table -->
  <section>
    <table mat-table [dataSource]="dataSource" matSort>
      <!-- Id Column -->
      <ng-container matColumnDef="id">
        <th mat-header-cell *matHeaderCellDef mat-sort-header>Id</th>
        <td mat-cell *matCellDef="let element">{{ element.id }}</td>
      </ng-container>

      <!-- Name Column -->
      <ng-container matColumnDef="name">
        <th mat-header-cell *matHeaderCellDef mat-sort-header>Name</th>
        <td mat-cell *matCellDef="let element">{{ element.name }}</td>
      </ng-container>

      <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
      <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
    </table>
  </section>

  <!-- Pagination -->
  <section>
    <mat-paginator [pageSizeOptions]="pageSizeOptions" showFirstLastButtons>
    </mat-paginator>
  </section>
</div>
