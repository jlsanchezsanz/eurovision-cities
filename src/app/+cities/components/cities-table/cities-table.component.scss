.wrapper {
  padding: 10px;
}

table {
  width: 100%;
}
