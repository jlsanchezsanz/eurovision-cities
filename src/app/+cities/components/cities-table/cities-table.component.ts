import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import { City } from '../../../models';

@Component({
  selector: 'app-cities-table',
  templateUrl: './cities-table.component.html',
  styleUrls: ['./cities-table.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CitiesTableComponent {
  @Input() public cities: City[];

  public displayedColumns: string[] = ['id', 'name'];
}
