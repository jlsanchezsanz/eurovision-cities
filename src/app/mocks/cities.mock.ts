import { CitiesResponse } from '../models';

export const citiesResponseMock: CitiesResponse = {
  content: [
    { id: 317, name: 'Ahvaz' },
    { id: 87, name: 'Al-Jizah' },
    { id: 147, name: 'Al-Khartum Bahri' },
    { id: 227, name: 'Al-Madinah' },
    { id: 103, name: 'Aleppo' },
    { id: 47, name: 'Alexandria' },
    { id: 176, name: 'Algiers' },
    { id: 307, name: 'Allahabad' },
    { id: 302, name: 'Almaty' },
    { id: 260, name: 'Amman' },
  ],
  pageable: {
    sort: { sorted: false, unsorted: true, empty: true },
    pageSize: 10,
    pageNumber: 1,
    offset: 10,
    unpaged: false,
    paged: true,
  },
  totalPages: 34,
  totalElements: 332,
  last: false,
  first: false,
  sort: { sorted: false, unsorted: true, empty: true },
  numberOfElements: 10,
  size: 10,
  number: 1,
  empty: false,
};
