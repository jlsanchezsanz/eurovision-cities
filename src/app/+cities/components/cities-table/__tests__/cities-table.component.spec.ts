import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import {
  MatTableModule,
  MatInput,
  MatFormField,
  MatPaginatorModule,
  MatSortModule,
} from '@angular/material';
import { MockComponent } from 'ng-mocks';

import { CitiesTableComponent } from '../cities-table.component';
import { citiesResponseMock } from '../../../../mocks';

describe('CitiesTableComponent', () => {
  let component: CitiesTableComponent;
  let fixture: ComponentFixture<CitiesTableComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        BrowserAnimationsModule,
        MatTableModule,
        MatPaginatorModule,
        MatSortModule,
      ],
      declarations: [
        CitiesTableComponent,
        MockComponent(MatFormField),
        MockComponent(MatInput),
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CitiesTableComponent);
    component = fixture.componentInstance;
    component.cities = citiesResponseMock.content;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should update filter on input', () => {
    component.applyFilter(' asdF');
    expect(component.dataSource.filter).toBe('asdf');
  });
});
