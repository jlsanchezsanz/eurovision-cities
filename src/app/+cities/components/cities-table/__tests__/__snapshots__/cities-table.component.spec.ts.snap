// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesTableComponent should match snapshot 1`] = `
<app-cities-table
  dataSource={[Function MatTableDataSource]}
  displayedColumns={[Function Array]}
  paginator={[Function MatPaginator]}
  sort={[Function MatSort]}
>
  <div
    class="wrapper mat-elevation-z8"
  >
    <section>
      <mat-form-field>
        <input
          matinput=""
          placeholder="Filter"
        />
      </mat-form-field>
    </section>
    <section>
      <table
        class="mat-table"
        mat-table=""
        matsort=""
        role="grid"
      >
        <thead
          role="rowgroup"
        >
          
          <tr
            class="mat-header-row"
            mat-header-row=""
            role="row"
          >
            
            <th
              class="mat-header-cell cdk-column-id mat-column-id"
              mat-header-cell=""
              mat-sort-header=""
              role="columnheader"
            >
              Id
            </th>
            <th
              class="mat-header-cell cdk-column-name mat-column-name"
              mat-header-cell=""
              mat-sort-header=""
              role="columnheader"
            >
              Name
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              317
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Ahvaz
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              87
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Al-Jizah
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              147
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Al-Khartum Bahri
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              227
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Al-Madinah
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              103
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Aleppo
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              47
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Alexandria
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              176
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Algiers
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              307
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Allahabad
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              302
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Almaty
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-id mat-column-id"
              mat-cell=""
              role="gridcell"
            >
              260
            </td>
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
              Amman
            </td>
          </tr>
        </tbody>
        <tfoot
          class="mat-table-sticky"
          role="rowgroup"
          style="display: none; bottom: 0px; position: sticky; z-index: 10;"
        >
          
        </tfoot>
      </table>
    </section>
    <section>
      <mat-paginator
        showfirstlastbuttons=""
      />
    </section>
  </div>
</app-cities-table>
`;
